ibllib/io/video.py

```diff
@@ -31,25 +31,31 @@ def get_video_frame(video_path, frame_number):
 
 
 def get_video_frames_preload(video_path, frame_numbers=None, mask=Ellipsis, as_list=False,
-                             func=lambda x: x,):
+                             func=lambda x: x, quiet=False):
     """
     Obtain numpy array corresponding to a particular video frame in video_path.
     Fetching and returning a list is about 33% faster but may be less memory controlled. NB: Any
     gain in speed will be lost if subsequently converted to array.
-    :param video_path: URL or local path to mp4 file
+    :param video_path: URL or local path to mp4 file or cv2.VideoCapture object or
+                              oneibl.stream.VideoStreamer object
     :param frame_numbers: video frames to be returned. If None, return all frames.
     :param mask: a logical mask or slice to apply to frames
     :param as_list: if true the frames are returned as a list, this is faster but may be less
     memory efficient
     :param func: Function to be applied to each frame. Applied after masking if applicable.
+    :param quiet: if true, suppress frame loading progress output.
     :return: numpy array corresponding to frame of interest, or list if as_list is True.
     Default dimensions are (n, w, h, 3) where n = len(frame_numbers)
 
     Example - Load first 1000 frames, keeping only the first colour channel:
         frames = get_video_frames_preload(video_path, range(1000), mask=np.s_[:, :, 0])
     """
-    is_url = isinstance(video_path, str) and video_path.startswith('http')
-    cap = VideoStreamer(video_path).cap if is_url else cv2.VideoCapture(str(video_path))
+    is_cap = isinstance(video_path, (cv2.VideoCapture, VideoStreamer))
+    if is_cap:
+        cap = video_path
```

Comment:
Could we change the input name to 'vid'?  It's more accurate.  I'm not sure this will work... if video_path is an instance of VideoStreamer you need to get the cap handle from the cap property, something like:
```
if isinstance(video_path, cv2.VideoCapture):
    cap = video_path
elif  isinstance(video_path, VideoStreamer):
    cap = video_path.cap
else:
        is_url = isinstance(video_path, str) and video_path.startswith('http')
        cap = VideoStreamer(video_path).cap if is_url else cv2.VideoCapture(str(video_path))
```

Reply:
> Could we change the input name to 'vid'? 

Ah you don't mind changing the argument name? Sure I can do that. I was originally thinking "video_path_or_cap", but "vid" is simpler :)

> if video_path is an instance of VideoStreamer you need to get the cap handle from the cap property

How about just allowing a cv2.VideoCapture instance? For example in the motion energy function I can use this same line
`cap = VideoStreamer(video_path).cap if is_url else cv2.VideoCapture(str(video_path))`
and then just input the result to preload_video_frames

Reply:
> How about just allowing a cv2.VideoCapture instance? 
Only your function provides a cv2.VideoCapture object because it's the only one that calls this function many times over.  If you narrow the input types then you have refactor everyone else's code and we'll end up with more lines overall because everyone will have to open their videos within the calling function.  The if-elif-else statement in the comment above should cover all current input types, including a VideoCapture handle.

Reply:
No I think that was a misunderstanding. I wasn't suggesting to remove the option of inputting a Path of course. Just the option of inputting an VideoStreamer instance. My suggestion (as implemented in the last version) is to allow either a local path, a URL, or a VideoCapture instance as input. If one wants to create a VideoStreamer instance outside of the function they can create it and then pass VideoStreamer.cap.  That shouldn't break anyones code as it only adds functionality